services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,418 @@
+        }
+
+        and: "Flow is pingable"
+        def responsePing1 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing1.forward.pingSuccess
+        responsePing1.reverse.pingSuccess
+
+        when: "Try to update the encapsulation type to #encapsulationUpdate.toString()"
+        northbound.updateFlow(flow.id, flow.tap { it.encapsulationType = encapsulationUpdate })
+
+        then: "The encapsulation type is changed to #encapsulationUpdate.toString()"
+        def flowInfo2 = northbound.getFlow(flow.id)
+        flowInfo2.encapsulationType == encapsulationUpdate.toString().toLowerCase()
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.id).each { direction -> assert direction.asExpected }
+
+        and: "The flow allows traffic"
+        [exam.forward, exam.reverse].each { direction ->
+            def resources = traffExam.startExam(direction)
+            direction.setResources(resources)
+            assert traffExam.waitExam(direction).hasTraffic()
+        }
+
+        and: "Flow is pingable"
+        def responsePing2 = northbound.pingFlow(flow.id, new PingInput())
+        responsePing2.forward.pingSuccess
+        responsePing2.reverse.pingSuccess
+
+        //TODO(andriidovhan) check rules(tunnel-id) when pr2503 is merged
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        where:
+        encapsulationCreate | encapsulationUpdate
+        FlowEncapsulationType.TRANSIT_VLAN | FlowEncapsulationType.VXLAN
+        FlowEncapsulationType.VXLAN | FlowEncapsulationType.TRANSIT_VLAN
+    }
+
+    @Unroll
+    @Tags(HARDWARE)
+    def "Able to CRUD #flowDescription pinned flow"() {
+        when: "Create a flow"
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { it.src.noviflow && it.dst.noviflow }
+        assumeTrue("Unable to find required switches in topology", switchPair as boolean)
+
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.maximumBandwidth = bandwidth
+        flow.ignoreBandwidth = bandwidth == 0
+        flow.encapsulationType = FlowEncapsulationType.VXLAN
+        flow.pinned = true
+        flowHelper.addFlow(flow)
+
+        then: "Pinned flow is created"
+        def flowInfo = northbound.getFlow(flow.id)
+        flowInfo.pinned
+
+        when: "Update the flow (pinned=false)"
+        northbound.updateFlow(flow.id, flow.tap { it.pinned = false })
+
+        then: "The pinned option is disabled"
+        def newFlowInfo = northbound.getFlow(flow.id)
+        !newFlowInfo.pinned
+        flowInfo.lastUpdated < newFlowInfo.lastUpdated
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        where:
+        flowDescription | bandwidth
+        "a metered"     | 1000
+        "an unmetered"  | 0
+    }
+
+    @Unroll
+    @Tags(HARDWARE)
+    def "Able to CRUD a flow with protected path when maximumBandwidth=#bandwidth, vlan=#vlanId"() {
```

Comment:
Please say a word that this is 'vxlan' somewhere in test name

Reply:
I'm not sure, but how can you describe a vxlan-focused test without using 'vxlan' word?